core/corehttp/gateway_handler.go

```diff
@@ -81,7 +81,7 @@ func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string,
 	if strings.HasPrefix(p, IpnsPathPrefix) {
 		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
 		hash := elements[0]
-		rp, err := i.node.Namesys.Resolve(ctx, hash)
+		rp, err := i.node.Namesys.Resolve(ctx, hash, 0)
```

Comment:
We'll have to provide recommended bounds in the specs. this is a protocol problem, not an implementation one. The resolution limit picked will impact all sorts of implementations out there. If some implementation works with limit `k`, but others with limit `k/2`, it will create confusion for users. (dns referral / name recursion had a similar problem)